rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 2*time.Minute, "Maximum time a query may take before being aborted.")
+)
+
+var queryTimeoutErr = fmt.Errorf("query timeout after %v", *queryTimeout)
```

Comment:
Since var's are initialized before the flags are evaluated, this will always yield 2m0s instead of the actual timeout passed in via the flag.

So yes, the error string has to be created dynamically.